tests/zfs-tests/tests/functional/cli_root/zhack/library.kshlib

```diff
@@ -0,0 +1,293 @@
+#!/bin/ksh
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2021 by vStack. All rights reserved.
+#
+
+. "$STF_SUITE"/include/libtest.shlib
+. "$STF_SUITE"/include/blkdev.shlib
+
+#
+# Description:
+#
+# Test whether zhack label repair commands can recover detached devices
+# and corrupted checksums with a variety of sizes, and ensure
+# the purposes of either command is cleanly separated from the others.
+#
+# Strategy:
+#
+# Tests are done on loopback devices with sizes divisible by label size and sizes that are not.
+#
+# Test one:
+#
+# 1. Create pool on a loopback device with some test data
+# 2. Export the pool.
+# 3. Corrupt all label checksums in the pool
+# 4. Check that pool cannot be imported
+# 5. Verify that it cannot be imported after using zhack label repair -u
```

Comment:
This seems fine, but I'm not sure what this test step really buys us.

Reply:
The rationale for the extra tests was to make entirely sure the functionality of the two options was separate. `-u` should quit after seeing that checksums are corrupted. `-c` should not undetach a device.

I think it helped me during development, though at the same time, they probably aren't truly necessary now.

Reply:
That makes sense.  I just found it a bit confusing when reading through what the test was supposed to do.  I'm fine keeping it, I'd just reword the comment a bit to make it clear why you're doing this unnecessary thing.